### PR TITLE
refactor: use the auth config service to load the config from Kubernetes secrets

### LINKED ISSUE #6036: Use the auth config service to load the config from k8s secrets
### Summary 

It seems that right now some commands are loading directly the secrets from kubernetes secrets instead of relying on the factory abstraction for auth config service. This abstraction should handled when to load the config from kubernetes secrets, vault or local file. 

The code require some refactoring to move the kubernetes secrets loading into the auth config service.

### Jx version

The output of `jx version` is:

```
COPY OUTPUT HERE
```

### Jenkins type

- [ ] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### PR DESCRIPTION
**Pull request by ccojocar**
* Replace the direct load of auth config from Kubernetes secrets with auth config service through the code
* Adapt various test to use the auth config service instead of the Kubernetes secrets.
* Remove the methods from factory interface which loads the auth cofing directly from secrets, or creates the git providers from secrets

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #6036

### CODE REVIEW

File: pkg/auth/auth_config_test.go [hunk 1]
```diff
@@ -94,7 +85,7 @@ type ConfigTest struct {
 
 func (c *ConfigTest) Load() *auth.AuthConfig {
 	config, err := c.svc.LoadConfig()
-	c.AssertNoError(err)
+	require.NoError(c.t, err)
```
Comment (hferentschik) [inline pkg/auth/auth_config_test.go]: OOI, what's the difference between `assert` and `require`?

**Comment by hferentschik** [comment]
What would be the core change/file to look at?

File: pkg/auth/auth_config_test.go (hunk 1 above)
Comment (ccojocar) [reply pkg/auth/auth_config_test.go]: `require` stops immediately the tests without executing the follow up tests while `aseert` still keeps the test execution till to the end.

**Comment by ccojocar** [comment]
> What would be the core change/file to look at?

Please look at all files :). Thanks!

**Comment by garethjevans** [comment]
looks good, I can't spot anything obvious - keen to see how this will work when we don't have a vault ingress.

**Comment by ccojocar** [comment]
> looks good, I can't spot anything obvious - keen to see how this will work when we don't have a vault ingress.

At the moment the factory will look up for secrets sources in this order (and will not fail anymore if e.g. secrets are not found in vault):
1) if vault configured tries to load the secret from vault
2) load secrets from k8s secrets when the step 2) fails or vault is not configured
3) load the secrets from files either from `~/.jx` or `git/credentials` when step 2) fails 

We would most likely tweak the loading sequence but for the initial refactoring, this will ensure that everything keeps working as before.

**Pull request merged**